### ISSUE TITLE
Errors in class constructor silenced if function annotated to return instance of class

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1660]
**Describe the bug**
Errors in the constructor call of a class are silenced if a function is annotated to return an instance of the class and the class is instantiated in the return statement of the function.  Regression introduced in 1.1.122.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That line 8 would produce the same error as line 12.

**Screenshots or Code**
```python
# pyright: strict

class foo:
    def __init__(self, value: str) -> None: ...


def bar() -> foo:
    return foo(fdsfd)  # No error


def baz():
    return foo(fdsfd)  # '"fdsfd" is not defined'

```

**VS Code extension or command-line**
1.1.123

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release. Apologies for the regression.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.124, which I just published. It will also be included in the next release of pylance.
